arrow-data/src/main/kotlin/arrow/data/Try.kt

```diff
@@ -119,20 +120,16 @@ sealed class Try<out A> : TryKind<A> {
     @Deprecated(DeprecatedUnsafeAccess, ReplaceWith("getOrElse { ifEmpty }"))
     abstract fun get(): A
 
-    @Deprecated(DeprecatedUnsafeAccess, ReplaceWith("map { body(it); it }"))
-    fun onSuccess(body: (A) -> Unit): Try<A> {
-        foreach(body)
-        return this
+    fun onSuccess(body: (A) -> Unit): Try<A> = map {
```

Comment:
Given `body` is not actually captured and evaluating it may throw an exception I'd recommend we change the implementation to something like:
```kotlin
fun onSuccess(body: (A) -> Unit): Try<A> = 
  flatMap { Try { body(it); it } }
```
This ensures that if `body` is ill behaved any thrown exceptions would be represented as a `Failure` and this function will not blow up at run-time. Here is an example that would make the current implementation fail with an exception:
```kotlin
Try { "x "}.onSuccess { println(it.toInt()) }
```
In my opinion these functions belong in `import arrow.syntax.try.unsafe.*` because the function itself is not referentially transparent because for different inputs it produces the same observable output and hidden chain of side effects.